src/bpftrace.cpp

```diff
@@ -1305,6 +1305,21 @@ std::string BPFtrace::map_value_to_str(const SizedType &stype,
 
     return "[" + str_join(elems, ",") + "]";
   }
+  else if (stype.IsRecordTy())
```

Comment:
Should we do something fancy for the JSON output? Or just leave as is?

Reply:
We could replace "=" by ":" and the result string would be a valid JSON (although there would be no type information). Alternatively, we should at least put the value into quotes (and same for arrays I think).

Reply:
Yeah, at the very least it should be quoted. For example, this is currently an issue:
```
$ sudo ./build/src/bpftrace -e 'struct Foo { int x; int y; } BEGIN { $s = *(struct Foo *)0; print($s) }' -f json
{"type": "attached_probes", "data": {"probes": 1}}
{"type": "value", "data": { .x = 0, .y = 0 }}
^C

$ sudo ./build/src/bpftrace -e 'struct Foo { int x; int y; } BEGIN { $s = *(struct Foo *)0; print($s) }' -f json | jq
{
  "type": "attached_probes",
  "data": {
    "probes": 1
  }
}
parse error: Invalid numeric literal at line 2, column 31
^C% 
```

Reply:
I think it'd make sense to make struct output be a valid json object. ie

```
{
  "x": 0,
  "y": 0
}
```

Reply:
Ok, I'll do that.

Reply:
So I have a rather simple solution (4287e7b) that works for basic cases, but has problems e.g. for structs nested in tuples (because the quotes around field names get escaped). 

Handling all the cases would require a more complex refactoring of the `Output` class and I won't get to it until the next week. So if you think this is sufficient for the moment, go on and merge the PR and I'll fix the JSON outputs properly in a new one.

Reply:
The output handling is quite ugly atm, spread over a lot of parts and messy to follow. A refactoring of the whole output logic would be nice. I wouldn't mind if the output code is more library like where we have a common interface text,json and other output formats can use. But thats quite a rewrite.

Maybe we can warn/error when nested struct and json are used, stating its still WIP. 

What do you think @danobi. Would be great to get this in.

Reply:
I added a warning when a "struct inside tuple" is printed to JSON. Should be now ready to merge.